TipCatDotNet.Api/Services/Payments/TransactionService.cs

```diff
@@ -138,33 +138,13 @@ Result Validate()
 
 
         async Task<Result<List<FacilityTransactionResponse>>> GetTransactions()
-        {
-            var now = DateTime.UtcNow;
-
-            var facilities = await _context.Facilities
-                .Where(f => f.AccountId == accountId)
-                .Select(FacilityProjection())
-                .ToListAsync(cancellationToken);
-
-            var transactions = await _context.Transactions
-                .GroupBy(t => new { t.FacilityId, t.Currency })
-                .Select(x => new
-                {
-                    Keys = x.Key,
-                    Total = x.Sum(x => x.Amount)
-                })
-                .ToDictionaryAsync(x => x.Keys.FacilityId,
-                    x => new MoneyAmount(x.Total, (Currencies)Enum.Parse(typeof(Currencies), x.Keys.Currency, true)));
-
-            var result = new List<FacilityTransactionResponse>(facilities.Count);
-            foreach (var facility in facilities)
-            {
-                transactions.TryGetValue(facility.Id, out var totalAmount);
-                result.Add(new FacilityTransactionResponse(facility, totalAmount));
-            }
-
-            return result;
-        }
+            => _context.Transactions
+                .Join(_context.Facilities, t => t.FacilityId, f => f.Id, FacilityTransactionProjection())
+                .AsEnumerable()
+                .Where(x => x.Facility.AccountId == accountId)
```

Comment:
Could you place `Where` before `AsEnumerable`? Otherwise you download all transactions in memory

Reply:
I've tried. But query fall in error:
could not be translated. Either rewrite the query in a form that can be translated, or switch to client evaluation 
explicitly by inserting a call to 'AsEnumerable', 'AsAsyncEnumerable', 'ToList', or 'ToListAsync'

Reply:
Could it be re-writed?